app/src/main/java/org/piramalswasthya/sakhi/helpers/CommonUtils.kt

```diff
@@ -312,6 +314,103 @@ fun filterBenHRPTFormList(
     }
 }
 
+
+fun filterImmunList(list: List<ImmunizationDetailsDomain>, text: String): List<ImmunizationDetailsDomain> {
+    if (text == "")
+        return list
+    else {
+        var filterText = text.lowercase()
+        var firstVal = ""
+        var secondVal = ""
+        var thirdVal = ""
+        var fourthVal = ""
+        var fifthVal = ""
+        var sixthVal = ""
+        var seventhVal = ""
+        var eighthVal = ""
+
+        if (filterText.contains("5-6 years")) {
+            firstVal =   "${filterText.split("-")[0]} years"
+            filterText = filterText.split("-")[1]
+
+        } else if (filterText.contains("16-24 months")) {
+            firstVal = "1 years"
+            /*secondVal = "17 months"
+            thirdVal = "18 months"
+            fourthVal = "19 months"
+            fifthVal = "20 months"
+            sixthVal = "21 months"
+            seventhVal = "22 months"
+            eighthVal = "23 months"*/
+            filterText = "2 years"
+
+        }
+        else if (filterText.contains("9-12 months")) {
+            firstVal = "${filterText.split("-")[0]} months"
+            secondVal = "10 months"
+            thirdVal = "11 months"
+            filterText = filterText.split("-")[1]
+
+        }
+        else if (filterText.contains("6 weeks")) {
+            firstVal = "1 months"
+            filterText = "2 months"
+
+        }
+        else if (filterText.contains("10 weeks")) {
+
+            filterText = "3 months"
+
+        }
+        else if (filterText.contains("14 weeks")) {
+
+            filterText = "4 months"
+
+        }
+        return list.filter {
+
+            filterForImm(
```

Comment:
What are these "val" variables? Doesn't seem to be intuitive.

Reply:
Sorry and  I have changed this and also removed all the unnecessary comment and line breaks.